src/nvim/buffer_updates.c

```diff
@@ -289,7 +289,12 @@ void buf_updates_send_splice(
     int new_row, colnr_T new_col, bcount_t new_byte)
 {
   if (!buf_updates_active(buf)
-      || (old_byte == 0 && new_byte == 0)) {
+      || (old_row == 0
```

Comment:
this a bit weird, ideally we want to assert that `old_byte == 0` iff `old_row == 0 && old_col == 0`  (and `:s/old/new/g` )

Reply:
though we can do this in the test layer, to avoid unpleasant crashes for now (can escalate later)

Reply:
Oh I see, removing from this PR then